src/drivers/rpi_rc_in/rpi_rc_in.cpp

```diff
@@ -0,0 +1,224 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+#include "rpi_rc_in.h"
+
+using namespace rpi_rc_in;
+//---------------------------------------------------------------------------------------------------------//
+int RcInput::rpi_rc_init()
+{
+	int i;
+
+	//--------------初始化共享内存映射----------------------------//
+	if ((this->shmid = shmget(this->key, sizeof(int) * this->_channels, 0666))
+	    < 0) {
+		PX4_WARN("无法访问共享内存。Faild to access shared memory");
+		return -1;
+	}
+
+	if ((this->mem = (int *) shmat(this->shmid, NULL, 0)) == (void *) - 1) {
+		PX4_WARN("无法映射共享内存。Faild to mapping shared memory");
+		return -1;
+	}
+
+	//--------------发布所有通道的数据------------------------//
+	for (i = 0; i < input_rc_s::RC_INPUT_MAX_CHANNELS; ++i) {
+		_data.values[i] = UINT16_MAX;
+	}
+
+	_rcinput_pub = orb_advertise(ORB_ID(input_rc), &_data);
+
+	if (_rcinput_pub == nullptr) {
+		PX4_WARN("error: advertise failed");
+		return -1;
+	}
+
+	return 0;
+}
+//---------------------------------------------------------------------------------------------------------//
+int RcInput::start()
+{
+	int result = 0;
+
+	result = rpi_rc_init();
+
+	if (result != 0) {
+		PX4_WARN("error: RC initialization failed");
+		return -1;
+	}
+
+	_isRunning = true;
+	result = work_queue(HPWORK, &_work, (worker_t) & RcInput::cycle_trampoline,
+			    this, 0);
+
+	if (result == -1) {
+		_isRunning = false;
+	}
+
+	return result;
+}
+//---------------------------------------------------------------------------------------------------------//
+void RcInput::stop()
+{
+	_shouldExit = true;
+}
+//---------------------------------------------------------------------------------------------------------//
+void RcInput::cycle_trampoline(void *arg)
+{
+	RcInput *dev = reinterpret_cast<RcInput *>(arg);
+	dev->_cycle();
+}
+//---------------------------------------------------------------------------------------------------------//
+void RcInput::_cycle()
+{
+	_measure();
+
+	if (!_shouldExit) {
+		work_queue(HPWORK, &_work, (worker_t) & RcInput::cycle_trampoline, this,
+			   USEC2TICK(RCINPUT_MEASURE_INTERVAL_US));
+	}
+}
+//---------------------------------------------------------------------------------------------------------//
+void RcInput::_measure(void)
+{
+	uint64_t ts;
+	// PWM数据发布
+	// read pwm value from shared memory
+	int i = 0;
+
+	for (i = 0; i < _channels; ++i) {
+		_data.values[i] = (*(this->mem + i) <= 0) ? UINT16_MAX : *(this->mem + i);
```

Comment:
I found that if you place the throttle stick on the top of the position,  some motors will stop.
I referred to the driver of navio, they treat the problem as same.
But it's not the issue of the driver,the param of QGC should be changed as following line
1. "PWM DISARMED" and "PWM AUX DISARMED"   should be less than the minimum pwm of your throttle
2. "PWM MAX" and "PWM AUX MAX" should be less than the maxmum pwm of your throttle,
 the maxmum value minus 100 was recommend